src/time.cr

```diff
@@ -181,7 +181,7 @@ struct Time
   # ```
   # Time.epoch(981173106) # => 2001-02-03 04:05:06 UTC
   # ```
-  def self.epoch(seconds : Int) : self
+  def self.epoch(seconds : Int) : Time
```

Comment:
Why not keep `self` here?

Reply:
1. If something subclasses `Time`, this will not hold anymore (uhh subclassing a `struct` is impossible, but the theoretic point still stands)
2. It can be confused with actually returning `self` (but modified). This way it's clear that it's an entirely new instance.

Reply:
1. hmm I'm not sure I understand, do you think that if one subclasses Time to Time2 (in theory), the returned type of `Time.epoch` could be understood as `Time2` ? Is that what you're trying to avoid? (TIL something! I thought it was just a shortcut to the current class)
2. Well, it is a class-level method, so it won't be able to 'return' `self` in any case.

Also, I think the doc generator should change `self` to the actual type (when it's used as a type) when possible & known.

Reply:
Actually, since `epoch` calls `new`, which can be overloaded by the theoretical subclass (if `s/struct/class/`), it could return anything!

It's theoretical, `Time` is fine.